wrappers/services.go

```diff
@@ -224,6 +224,34 @@ func StopServices(apps []*snap.AppInfo, inter interacter) error {
 		if !app.IsService() || !osutil.FileExists(app.ServiceFile()) {
 			continue
 		}
+		// Skip stop on refresh when refresh mode is "keep"
+		if reason == snap.StopReasonRefresh {
+			switch app.RefreshMode {
+			case "keep":
+				// skip this service
+				continue
+			case "sigterm":
+				sysd.Kill(app.ServiceName(), "TERM", "main")
+				continue
+			case "sigterm-all":
+				sysd.Kill(app.ServiceName(), "TERM", "all")
+				continue
+			case "sighup":
+				sysd.Kill(app.ServiceName(), "HUP", "main")
+				continue
+			case "sighup-all":
+				sysd.Kill(app.ServiceName(), "HUP", "all")
+				continue
+			case "sigusr1":
```

Comment:
Any reason to not support usr2 as well? Those are the typical signals I can think of for implementing custom behavior (hup, usr1, usr2).. any others?

Reply:
Sure, I added sigusr2 now as well. I think (hup,usr1,usr2) are the ones used. We can easily add more if needed but it seems the other ones are really exotic (or inappropriate).